Be more strict when looking for the desired VS

cmb69 opened the pull request:
On systems where VS 2017 and VS 2019 are available, the starter script
may choose VS 2019 if vc15 is requested.  The documentation on vswhere
isn't particularly clear regarding the -version parameter, but it
states that a version range is expected.  Therefore we change the
shell setup script accordingly.

cmb69 committed:
Be more strict when looking for the desired VS

On systems where VS 2017 and VS 2019 are available, the starter script
may choose VS 2019 if vc15 is requested.  The documentation on vswhere
isn't particularly clear regarding the -version parameter, but it
states that a version range is expected.  Therefore we change the
shell setup script accordingly.

cmb69 commented:
Example session on my machine:
````
D:\php-sdk\bin>vswhere -nologo -version 15 -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath -format text
installationPath: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community

installationPath: C:\Program Files (x86)\Microsoft Visual Studio\2017\Community

D:\php-sdk\bin>vswhere -nologo -version [15,16) -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath -format text
installationPath: C:\Program Files (x86)\Microsoft Visual Studio\2017\Community
````

weltling commented on bin/phpsdk_setshell.bat:
```diff
@@ -81,21 +81,24 @@ if 15 gtr %PHP_SDK_VS_NUM% (
+	set /a PHP_SDK_VS_RANGE=PHP_SDK_VS_NUM + 1
+	set PHP_SDK_VS_RANGE="[%PHP_SDK_VS_NUM%,!PHP_SDK_VS_RANGE%!)"
+
+	for /f "tokens=1* delims=: " %%a in ('%~dp0\vswhere -nologo -version !PHP_SDK_VS_RANGE! -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath -format text') do (
```
Great idea with the range. With this, the breaks are not needed in this loop, as it delivers just one result. Perhaps it would make sense to get rid of these breaks? The reason I rewrote it is that on my side the order was reversed - first 2017, then 2019. Now with your approach it seems to be consistent.

Thanks.

cmb69 replied:
Ah, yeah. I wonder what would happen if one has a preview and an RC installed side-by-side, though. In that case likely two installations will be found, i.e. two lines.

weltling replied:
Nope, preview is checked further down the line, attaching `-prerelease` to the command. Until `vswhere` changes behaviors, it should be ok.

Thanks.

cmb69 committed:
Remove break constructs

These had been introduced to cater to multiple found VS installations,
to only use the first one.  Since the ranges are supposed to find only
a single installation, and since apparently the order of multiple
findings is not guaranteed, we remove the break constructs for better
readability of the source code.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/php-sdk-binary-tools?pullRequest=56) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/php-sdk-binary-tools?pullRequest=56) before we can accept your contribution.<br/><br/>:x: cmb69 [sign now](https://cla.opensource.microsoft.com/Microsoft/php-sdk-binary-tools?pullRequest=56)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/php-sdk-binary-tools?pullRequest=56) it.</sub>
